core/src/Kokkos_Core_fwd.hpp

```diff
@@ -34,11 +34,12 @@
 #endif
 
 //----------------------------------------------------------------------------
-// Have assumed a 64bit build (8byte pointers) throughout the code base.
-
+// Have assumed a 64-bit build (8-byte pointers) throughout the code base.
+// 32-bit build allowed but unsupported.
+#ifndef KOKKOS_IMPL_32BIT
 static_assert(sizeof(void *) == 8,
               "Kokkos assumes 64-bit build; i.e., 8-byte pointers");
-
+#endif
```

Comment:
Did you consider:
* having an assertion that it is equal to 4 in 32-bit mode
* or just getting rid of it